python/ngen_cal/src/ngen/cal/agent.py

```diff
@@ -72,17 +77,14 @@ def __init__(self, model_conf, workdir: Path, log: bool=False, restart: bool=Fal
             # 0 correctly since not all basin params can be loaded.
             # There are probably some similar issues with explicit and independent, since they have
             # similar data semantics
-            workdirs = list(Path.glob(workdir, model_conf['type']+"_*_worker"))
+            workdirs = list(Path.glob(workdir, ngen_model.type+"_*_worker"))
             if len(workdirs) > 1:
                 print("More than one existing workdir, cannot restart")
             elif len(workdirs) == 1:
-                self._job = JobMeta(model_conf['type'], workdir, workdirs[0], log=log)
+                self._job = JobMeta(ngen_model.type, workdir, workdirs[0], log=log)
 
         if self._job is None:
-            self._job = JobMeta(model_conf['type'], workdir, log=log)
-        resolved_binary = Path(model_conf['binary']).resolve()
-        model_conf['workdir'] = self.job.workdir
-        self._model = Model(model=model_conf, binary=resolved_binary)
+            self._job = JobMeta(ngen_model.type, workdir, log=log)
```

Comment:
I think we are still missing the updated model workdir.  An agent has a job which nests information and job/model files under a dynamically created workdir `job.workdir`.  Any model instance associated with that `job` must have its `workdir` aligned with the `job.workdir`, so we need to update the model reference like such
```suggestion
            self._job = JobMeta(ngen_model.type, workdir, log=log)
        ngen_model.workdir = self.job.workdir
```